charger/ocpp.go

```diff
@@ -165,107 +170,121 @@ func NewOCPP(id string, connector int, idtag string,
 	case <-cp.HasConnected():
 	}
 
-	// see who's there
-	if boot {
-		conn.TriggerMessageRequest(core.BootNotificationFeatureName)
-	}
-
 	var (
-		rc                  = make(chan error, 1)
-		meterSampleInterval time.Duration
-	)
+		rc = make(chan error, 1)
 
-	keys := []string{
-		ocpp.KeyNumberOfConnectors,
-		ocpp.KeyMeterValuesSampledData,
-		ocpp.KeyMeterValueSampleInterval,
-		ocpp.KeyConnectorSwitch3to1PhaseSupported,
-		ocpp.KeyChargingScheduleAllowedChargingRateUnit,
-	}
-	_ = keys
+		MeterValuesSampledData bool
+
+		// If a key value is defined as a CSL, it MAY be accompanied with a [KeyName]MaxLength key, indicating the
+		// max length of the CSL in items. If this key is not set, a safe value of 1 (one) item SHOULD be assumed.
+		MeterValuesSampledDataMaxLength int = 1
+	)
 
 	c.chargingRateUnit = types.ChargingRateUnitType(chargingRateUnit)
 
-	// noConfig mode disables GetConfiguration
-	if noConfig {
-		c.meterValuesSample = meterValues
-		if meterInterval == 0 {
-			meterInterval = 10 * time.Second
-		}
-	} else {
-		// fix timing issue in EVBox when switching OCPP protocol version
-		time.Sleep(time.Second)
-
-		err := ocpp.Instance().GetConfiguration(cp.ID(), func(resp *core.GetConfigurationConfirmation, err error) {
-			if err == nil {
-				// log unsupported configuration keys
-				if len(resp.UnknownKey) > 0 {
-					c.log.ERROR.Printf("unsupported keys: %v", resp.UnknownKey)
+	// fix timing issue in EVBox when switching OCPP protocol version
```

Comment:
Komplett in eigene Funktion raus ziehen?

Reply:
Was genau meinst du?